pkg/server/event_handler.go

```diff
@@ -293,6 +299,46 @@ func (h *EventHandler[T, P]) getGCSObjectProto(ctx context.Context, objAttrs map
 	return p, nil
 }
 
+func (h *EventHandler[T, P]) getGCSObjectGithubResourceMetadata(ctx context.Context, objAttrs map[string]string) (*v1alpha1.GitHubSource, error) {
+	bucketID, found := objAttrs["bucketId"]
+	if !found {
+		return nil, fmt.Errorf("bucket ID not found")
+	}
+	objectID, found := objAttrs["objectId"]
+	if !found {
+		return nil, fmt.Errorf("object ID not found")
+	}
+
+	// Read the object from bucket.
+	attrs, err := h.client.Bucket(bucketID).Object(objectID).Attrs(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("object attrs not found")
+	}
+	m := attrs.Metadata
+	r := &v1alpha1.GitHubSource{}
+	c, found := m["git-commit"]
+	if found {
+		r.Commit = c
+	}
+	rn, found := m["git-repo"]
```

Comment:
Where do we set `file_path`?

Reply:
Xiyue and I talked about this last week. Currently, in the `file-change` and `file-copy workflows`, `file_path` is not uploaded to metadata. https://github.com/abcxyz/pmap/blob/main/.github/workflows/snapshot-file-change.yml#L73-L78
Our idea is for current PR, we only add this metadata, and there will be a separate PR that updates the workflow files to include `file_path`.
By the way, what do you think if we use that PR to fix the renaming and add more fields?

Reply:
Sure. It'd probably be easier if you do the other PR first to address the renaming and add file path

Reply:
To include `filepath` requires more workload, I'd prefer we handle it in another PR. We will need to change some logic in https://github.com/abcxyz/pmap/blob/main/.github/workflows/snapshot-file-copy.yml. Currently https://github.com/google-github-actions/upload-cloud-storage does not keep the all parent path of file when it uploads file to the gcs. https://github.com/abcxyz/pmap/blob/main/.github/workflows/snapshot-file-copy.yml need to follow some logic in https://github.com/abcxyz/pmap/blob/41c46ce082d64d08396dd60bb3b5a48762f08bdd/.github/workflows/snapshot-file-change.yml#L46 to record the info of `file_path` when it uploads file to gcs

Reply:
sgtm. Maybe the rename could be piggybacked here?

Reply:
I will do the renaming, adding field `workflow run id`, `run attempt`, and include `file_path` in the next PR.